File: ch2/2-9.c
#include <stdio.h>

int bitcount(int x);

int main(void)
{
    int x = 500;
    printf("%d\n", bitcount(x));
    return 0;
}

int bitcount(int x)
{
    int b;
    for (b = 0; x != 0; x &= (x-1)) {
        b++;
    }
    return b;
}

File: gkr/ch2/2-9.go
package main

import "fmt"

func main() {
	n := 75
	fmt.Printf("num of 1's in %d = %d\n", n, bitcount(n))
}

func bitcount(x int) int {
	var b int
	for b = 0; x != 0; x &= (x - 1) {
		if x > 0 {
			b++
		}
	}
	return b
}
